apps/timesheet-app/backend/modules/database/db_functions.bal

```diff
@@ -15,48 +15,129 @@
 // under the License.
 import ballerina/sql;
 
-# Fetch sample collections.
+# Fetch work policy of a company.
 #
-# + name - Name to filter  
-# + 'limit - Limit of the response  
-# + offset - Offset of the number of sample collection to retrieve
-# + return - List of sample collections|Error
-public isolated function fetchSampleCollections(string? name, int? 'limit, int? offset) returns SampleCollection[]|error {
-    stream<SampleCollection, error?> resultStream = databaseClient->
-            query(getSampleCollectionsQuery(name, 'limit, offset));
+# + companyName - Company name to filter
+# + return - Work policy or an error
+public isolated function fetchWorkPolicy(string companyName) returns WorkPolicy|error? {
+    WorkPolicy|error policies = databaseClient->queryRow(fetchWorkPolicyQuery(companyName));
 
-    SampleCollection[] sampleCollections = [];
-    check from SampleCollection sampleCollection in resultStream
-        do {
-            sampleCollections.push(sampleCollection);
-        };
+    if policies is sql:NoRowsError {
+        return;
+    }
+    return policies;
+}
 
-    return sampleCollections;
+# Fetch work policies of companies.
+#
+# + return - Work policies or an error
+public isolated function fetchWorkPolicies() returns WorkPolicy[]|error {
+    stream<WorkPolicy, error?> workPolicyResult = databaseClient->query(fetchWorkPolicyQuery(()));
+
+    return from WorkPolicy workPolicies in workPolicyResult
+        select workPolicies;
 }
 
-# Fetch specific sample collection.
+# Function to update work policy of a company.
 #
-# + id - Identification of the sample collection
-# + return - Sample collections|Error, if so
-public isolated function fetchSampleCollection(int id) returns SampleCollection|error? {
-    SampleCollection|sql:Error sampleCollection = databaseClient->queryRow(getSampleCollectionQuery(id));
+# + companyName - Name of the company to be updated
+# + workPolicies - Record to be updated
+# + invokerEmail - Email of the invoker
+# + return - An error if occurred
+public isolated function updateWorkPolicy(WorkPolicyUpdate workPolicies, string invokerEmail,
+        string companyName) returns error? {
 
-    if sampleCollection is sql:Error && sampleCollection is sql:NoRowsError {
-        return;
+    _ = check databaseClient->execute(updateWorkPolicyQuery(workPolicies, invokerEmail, companyName));
+}
+
+# Function to get timesheet records using filters.
+#
+# + filter - Filter type for the records
+# + return - TimeLog records or an error
+public isolated function fetchTimesheetRecords(TimesheetCommonFilter filter) returns TimeLog[]|error? {
```

Comment:
Since the function returns a `TimeLog[]` and the associated data table is `time_log`, consider renaming the function to reflect its purpose more accurately.

```suggestion
public isolated function fetchTimeLogs(TimeLogFilter filter) returns TimeLog[]|error? {
```